src/test/java/com/the/dark/side/crew/fejsbuk/service/impl/LikeServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+package com.the.dark.side.crew.fejsbuk.service.impl;
+
+import com.the.dark.side.crew.fejsbuk.mapper.LikeMapper;
+import com.the.dark.side.crew.fejsbuk.model.LikeEntity;
+import com.the.dark.side.crew.fejsbuk.model.PostEntity;
+import com.the.dark.side.crew.fejsbuk.model.UserEntity;
+import com.the.dark.side.crew.fejsbuk.model.dto.LikeDto;
+import com.the.dark.side.crew.fejsbuk.repository.LikeRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.web.server.ResponseStatusException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+class LikeServiceImplTest {
+
+    @InjectMocks
+    private LikeServiceImpl likeService;
+
+    @Mock
+    private LikeRepository likeRepository;
+
+    @Mock
+    private LikeMapper likeMapper;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void shouldAddLike() {
+        LikeDto likeDto = mock(LikeDto.class);
+        LikeEntity likeEntity = mock(LikeEntity.class);
+
+        when(likeMapper.toEntity(likeDto)).thenReturn(likeEntity);
+        when(likeMapper.toDto(likeEntity)).thenReturn(likeDto);
+
+        LikeDto result = likeService.addLike(likeDto);
+        verify(likeRepository).save(likeEntity);
+        assertEquals(result, likeDto);
+    }
+
+    @Test
+    void whenAlreadyLikedThen400() {
+        PostEntity postEntity = mock(PostEntity.class);
+        UserEntity userEntity = mock(UserEntity.class);
+        LikeEntity likeEntity = mock(LikeEntity.class);
+        LikeDto likeDto = mock(LikeDto.class);
+
+        when(likeEntity.getUserEntity()).thenReturn(userEntity);
+        when(likeEntity.getPostEntity()).thenReturn(postEntity);
+        when(likeMapper.toEntity(likeDto)).thenReturn(likeEntity);
+        when(likeRepository.existsByUserEntityAndPostEntity(userEntity, postEntity)).thenReturn(true);
+
+        ResponseStatusException exception =
+                assertThrows(ResponseStatusException.class, () -> likeService.addLike(likeDto));
+        assertEquals("400 BAD_REQUEST \"Already liked.\"", exception.getMessage());
+    }
+
+    @Test
+    void shouldRemoveLike() {
+        long likeId = 1L;
+        LikeEntity likeEntity = mock(LikeEntity.class);
+
+        when(likeEntity.getId()).thenReturn(likeId);
+
+        likeRepository.deleteById(likeEntity.getId());
+
```

Comment:
you can delete this line

Reply:
without this line test failed, any idea?

Reply:
line nr 74? :D